app/services/identity_reconciler.go

```diff
@@ -68,11 +73,37 @@ func IdentityReconciler(accountStore data.AccountStore, cfg *app.Config, provide
 	if err != nil {
 		return nil, errors.Wrap(err, "AccountCreator")
 	}
-	err = accountStore.AddOauthAccount(newAccount.ID, providerName, providerUser.ID, providerToken.AccessToken)
+	err = accountStore.AddOauthAccount(newAccount.ID, providerName, providerUser.ID, providerUser.Email, providerToken.AccessToken)
 	if err != nil {
 		// this should not happen since oauth details used to lookup account above
 		// not sure how best to test but feels appropriate to return error if encountered
 		return nil, errors.Wrap(err, "AddOauthAccount")
 	}
 	return newAccount, nil
 }
+
+func updateUserInfo(accountStore data.AccountStore, accountID int, providerName string, providerUser *oauth.UserInfo) error {
+	oAccounts, err := accountStore.GetOauthAccounts(accountID)
+	if err != nil {
+		return errors.Wrap(err, "GetOauthAccounts")
+	}
+
+	if len(oAccounts) == 0 {
+		return nil
+	}
+
+	for _, oAccount := range oAccounts {
+		if providerName != oAccount.Provider && providerUser.ID != oAccount.ProviderID {
+			continue
+		}
+
+		if oAccount.Email == "" {
```

Comment:
Is email changing a valid use case here?  Eg switching between apple private relay and "real" email address, or maybe changing email associated with a facebook account?  It may be easy enough to cover/test that we should include it even if unlikely.

Reply:
I'm not entirely certain about these use cases. Given that the queries take into account the provider ID, do you think it's significant?

Reply:
I'm not sure either but if the goal of capturing email is for end user communcation I think you want it to be right / up to date.